app/src/state/wallet.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import React, { useReducer, useContext, createContext, useEffect } from 'react'
+
+// Types
+import { WalletState, Dispatch, WalletActions } from './types'
+import { NetworkInfo, Registry } from '../constants/types'
+
+// Context
+import { useSwapContext } from '../context/swap.context'
+
+// Constants
+import { NATIVE_ASSET_CONTRACT_ADDRESS_0X } from '../constants/magics'
+
+// Create Wallet State Context
+const WalletStateContext = createContext<
+  { state: WalletState; dispatch: Dispatch } | undefined
+>(undefined)
```

Comment:
State and dispatch should be separate contexts
https://beta.reactjs.org/learn/scaling-up-with-reducer-and-context#step-2-put-state-and-dispatch-into-context

Reply:
Separated dispatch out into a `WalletStateDispatchContext`